e2eshark/tools/reportutil.py

```diff
@@ -43,67 +49,109 @@ def selectColumns(row, column_indices):
     return selectedcolumns
 
 
-def createMergedHeader(runnames, header):
-    mergedheader = ["test-name"]
+def createMergedHeader(args, runnames, header):
+    if args.mode == "summary":
+        mergedheader = ["items"]
+    else:
+        mergedheader = ["tests"]
     for i in range(len(header)):
         for run in runnames:
             columnname = header[i] + "-" + run
             mergedheader += [columnname]
     return mergedheader
 
 
-def getCanonicalizedListOfRuns(runnames, dictOfRuns, column_indices, rowlen):
+def getCanonicalizedListOfRuns(args, runnames, dictOfRuns, column_indices, rowlen):
     listOfRuns = []
     for run in runnames:
         if dictOfRuns.get(run):
             listOfRuns += [selectColumns(dictOfRuns[run], column_indices)]
         else:
-            listOfRuns += [["NA" for i in range(rowlen)]]
+            listOfRuns += [
+                ["NA" if args.mode == "status" else 0 for i in range(rowlen)]
+            ]
     return listOfRuns
 
 
-def createMergedRows(runnames, reportdict, column_indices, rowlen):
+def createOneMergedRow(
+    args, runnames, firstColumnIdentifier, dictOfRuns, column_indices, rowlen
+):
+    merged = [firstColumnIdentifier]
+    listOfRuns = getCanonicalizedListOfRuns(
+        args, runnames, dictOfRuns, column_indices, rowlen
+    )
+    # zip creates a tuple by taking same index value from each of the
+    # unpacked (using * operator) run in listOfRuns
+    for cellItemTuple in zip(*listOfRuns):
+        merged.extend(cellItemTuple)
+    return merged
+
+
+def createMergedRows(args, runnames, reportdict, column_indices, rowlen):
     mergedrows = []
     for test, dictOfRuns in reportdict.items():
-        merged = [test]
-        listOfRuns = getCanonicalizedListOfRuns(
-            runnames, dictOfRuns, column_indices, rowlen
+        merged = createOneMergedRow(
+            args, runnames, test, dictOfRuns, column_indices, rowlen
         )
-        # zip creates a tuple by taking same index value from each of the
-        # unpacked (using * operator) run in listOfRuns
-        for cellItemTuple in zip(*listOfRuns):
-            merged.extend(cellItemTuple)
-
         mergedrows += [merged]
 
     return mergedrows
 
 
-def createDiffRows(runnames, reportdict, column_indices, rowlen):
+def getDiff(args, tuple, diff):
+    # if it is a two element tuple, then provide exact difference
+    # for the pair for numbers, else say differ of match
+    if len(tuple) == 2:
```

Comment:
One thing. This makes perfect sense for `time`. But, for `summary` I was a little confused seeing the 0s and -1 for `total-count`. Maybe changing to "same" instead of the 0s can be easier to understand or just sticking with "same", [3,2]

Reply:
I see. I was hoping that it would be easier to spot how many tests stayed same (0) did not run or failed (-1) compared to yesterday or improved (+1). I added text as you suggested under -v:
```
The diff report for summary (time in seconds) for runs: fp32, bf16
| items        | tests    | model-run   | onnx-import   | torch-mlir     | iree-compile   | inference      |
|:-------------|:---------|:------------|:--------------|:---------------|:---------------|:---------------|
| total-count  | 0 (same) | 0 (same)    | 0 (same)      | -1 (regressed) | -1 (regressed) | -1 (regressed) |
| average-time | -0.234   | -0.132      | 0.044         | 0.003          | -0.141         | -0.008         |
| median-time  | 0.455    | 0.356       | 0.049         | 0.004          | 0.060          | -0.014         |
```